daemon/cmd/status.go

```diff
@@ -388,9 +388,19 @@ func (c *clusterNodesClient) NodeAdd(newNode nodeTypes.Node) error {
 
 func (c *clusterNodesClient) NodeUpdate(oldNode, newNode nodeTypes.Node) error {
 	c.Lock()
+	defer c.Unlock()
+
+	// If the node is on the added list, just update it
+	for i, added := range c.NodesAdded {
+		if added.Name == newNode.Fullname() {
+			c.NodesAdded[i] = newNode.GetModel()
+			return nil
```

Comment:
How do we remove IPs that were part of the old model in this case?

Broadly we track two IPs per node in this model: The node's IP for health-checking the node, then the health IP for the simulated health endpoint. From node A that is not changing, If a node B's Cilium agent restarts and triggers a node update across the cluster to update the cilium-health endpoint IP corresponding to this node, then I would expect that node A should receive that update, then pass the old IP and new IP through to the node A cilium-health agent so that it can follow the remove-then-add logic to ensure the old address is cleaned up:

https://github.com/cilium/cilium/blob/7457ce6299dc64f4b6a810fef8c8dd9f2bc4f4fa/pkg/health/server/prober.go#L189-L197

Reply:
At a glance it looks like with this change we will continue to ping the original health IP from node B forever.

Reply:
Ah, I think one piece I was missing in initial review is that in this particular case, the node being updated was not yet sent through to the health client yet at all, so there's nothing to remove.

The comment `If the node is on the added list, just update it` explains *what*, but not *why*. Studying `NodeDelete()` shows that this kind of code is specifically for handling the collection of updates for the next bulk node update to the cilium-health server, where presence in the `c.NodesAdded` slice implies changes that have not yet been propagated to the nodes client (ie cilium-health server).